src/classes/RD2_EntryFormController.cls

```diff
@@ -281,47 +290,108 @@ public with sharing class RD2_EntryFormController {
+        }
+
+        UTIL_Http.Response response = sendCommitmentRequest(recordId, jsonRequestBody);
+
+        saveCommitment(recordId, response);
+
+        return JSON.serialize(response);
+    }
+
+    /***
+    * @description Sends an HttpRequest so that matching Recurring Commitment is created in Elevate
+    * @param jsonRequestBody JSON containing parameters for the purchase call request body
+    * @return String: JSON containing the response with success/error status and message
+    */
+    public static UTIL_Http.Response sendCommitmentRequest(Id recordId, String jsonRequestBody) {
         UTIL_Http.Response response;
-        
+
         try {
-            response = sendCommitmentRequest(jsonRequestBody);
+            HttpRequest request = new PS_CommitmentRequest()
+                .buildCreateRequest(jsonRequestBody);
 
-            if (response.statusCode == UTIL_Http.STATUS_CODE_CREATED) {
-                // If the commitment is created, then update RD fields: 
-                // Commitment Id, credit card expiry date and last 4.
+            response = requestService.sendRequest(request);
 
-                // Ensure DML error (if any) on the RD update is returned to the LWC 
-                // as a response or an AuraHandled exception (either will work).
-            }        
-        
         } catch (Exception ex) {
             response = requestService.buildErrorResponse(ex);
 
         } finally {
-            // If the response contains an error, regardless if the commitment is created, or
-            // the RD update failed, create an error log
+            if (response.statusCode != UTIL_Http.STATUS_CODE_CREATED) {
+                logError(recordId, response);
+            }
         }
 
-        return JSON.serialize(response);
+        return response;
     }
 
     /***
-    * @description Sends an HttpRequest so that matching Recurring Commitment is created in Elevate
-    * @param jsonRequestBody JSON containing parameters for the purchase call request body
-    * @return String: JSON containing the response with success/error status and message
+    * @description Saves commitment Id and the credit card data onto the Recurring Donation record.
+    * @param recordId Recurring Donation Id
+    * @param response Response from Elevate on the commitment create request
+    * @return void
+    */
+    @TestVisible
+    private static void saveCommitment(Id recordId, UTIL_Http.Response response) {
+        if (response == null 
+            || response.statusCode != UTIL_Http.STATUS_CODE_CREATED
+        ) {
+            return;
+        }
+
+        try {
+            PS_CommitmentRequest.ResponseBody commitment = new PS_CommitmentRequest().getCommitment(response);
+
+            if (commitment.hasId()) {
+                npe03__Recurring_Donation__c rd = new npe03__Recurring_Donation__c(
+                    Id = recordId,
+                    CommitmentId__c = commitment.id,
```

Comment:
Looking at the unpackaged config, these two new fields are defined as Readonly in the Admin/Standard profile permissions - which is probably fine. However, this method that updates these fields is in a class defined as "With Sharing". I think these need to be updated in an inner class (somewhere) defined as "Without Sharing" (rather than grant edit access to all users).